eng/common/scripts/Verify-Links.ps1

```diff
@@ -0,0 +1,234 @@
+param (
+  # url to verify links. Can either be a http address or a local file request. Local file paths support md and html files.
+  [string[]] $urls,
+  # file that contains a set of links to ignore when verifying
+  [string] $ignoreLinksFile = "$PSScriptRoot/ignore-links.txt",
+  # switch that will enable devops specific logging for warnings
+  [switch] $devOpsLogging = $false,
+  # check the links recurisvely based on recursivePattern
+  [switch] $recursive = $true,
+  # recusiving check links for all links verified that begin with this baseUrl, defaults to the folder the url is contained in
+  [string] $baseUrl = "",
+  # path to the root of the site for resolving rooted relative links, defaults to host root for http and file directory for local files
+  [string] $rootUrl = "",
+  # list of http status codes count as broken links. Defaults to 404. 
+  [array] $errorStatusCodes = @(404)
+)
+
+$ProgressPreference = "SilentlyContinue"; # Disable invoke-webrequest progress dialog
+
+function NormalizeUrl([string]$url){
+  if (Test-Path $url) {
+    $url = "file://" + (Resolve-Path $url).ToString();
+  }
+
+  $uri = [System.Uri]$url;
+
+  if ($script:baseUrl -eq "") {
+    # for base url default to containing directory
+    $script:baseUrl = (new-object System.Uri($uri, ".")).ToString();
+  }
+
+  if ($script:rootUrl -eq "") {
+    if ($uri.IsFile) { 
+      # for files default to the containing directory
+      $script:rootUrl = $script:baseUrl;
+    }
+    else {
+      # for http links default to the root path
+      $script:rootUrl = new-object System.Uri($uri, "/");
+    }
+  }
+  return $uri
+}
+
+function LogWarning
+{
+  if ($devOpsLogging)
+  {
+    Write-Host "##vso[task.LogIssue type=warning;]$args"
+  }
+  else
+  {
+    Write-Warning "$args"
+  }
+}
+
+function ResolveUri ([System.Uri]$referralUri, [string]$link)
+{
+  $linkUri = [System.Uri]$link;
+
+  if (!$linkUri.IsAbsoluteUri) {
```

Comment:
Add a new option called "resolveRelativeLinks" set to true but in our usage we can pas in false in our yml template. Only run this resolution logic to convert to absolute uri when resolveRelativeLinks is set to true.